pybop/costs/base_cost.py

```diff
@@ -82,29 +85,20 @@ def __call__(self, inputs: Union[Inputs, list], calculate_grad: bool = False):
         ValueError
             If an error occurs during the calculation of the cost.
         """
-        if self.transformation:
-            p = self.transformation.to_model(inputs)
-        inputs = self.parameters.verify(p if self.transformation else inputs)
+        inputs = self.parameters.verify(inputs)
         self.parameters.update(values=list(inputs.values()))
         y, dy = None, None
 
-        try:
-            if self._has_separable_problem:
-                if calculate_grad is True:
-                    y, dy = self.problem.evaluateS1(self.problem.parameters.as_dict())
-                else:
-                    y = self.problem.evaluate(
-                        self.problem.parameters.as_dict(),
-                        update_capacity=self.update_capacity,
-                    )
-
-            return self.compute(y, dy=dy, calculate_grad=calculate_grad)
-
-        except NotImplementedError as e:
-            raise e
+        if self._has_separable_problem:
```

Comment:
I think the transformation should be applied within the cost class directly. This allows users to interact with the cost class with transformations if requested. One benefit of this is the user can create their own optimisation class, or investigate the transformed cost object directly. Was there a technical reason that this needed to be moved to the optimisation class?

Reply:
At present, keeping transformations within the optimiser provides a cleaner and more reliable experience for the user. It also allows the user to pass custom cost functions (that are not pybop.cost objects) to the optimiser and take advantage of all the same functionality. I would vote to keep transformations in the optimiser class.